keyboards/planck/keymaps/yhaliaw/keymap.c

```diff
@@ -0,0 +1,233 @@
+
+const rgblight_segment_t PROGMEM lower_layer[] = RGBLIGHT_LAYER_SEGMENTS({0, 8, HSV_RED});
+
+const rgblight_segment_t PROGMEM raise_layer[] = RGBLIGHT_LAYER_SEGMENTS({0, 8, HSV_GREEN});
+
+const rgblight_segment_t PROGMEM adjust_layer[] = RGBLIGHT_LAYER_SEGMENTS({0, 8, HSV_BLUE});
+
+const rgblight_segment_t PROGMEM capslock_layer[] = RGBLIGHT_LAYER_SEGMENTS({4, 5, HSV_WHITE});
+
+// Later layers take precedence.
+const rgblight_segment_t* const PROGMEM rgb_layers[] = RGBLIGHT_LAYERS_LIST(
+    base_layer,
+    lower_layer,
+    raise_layer,
+    adjust_layer,
+    capslock_layer
+);
+
+bool audio_disable = false;
+
+void keyboard_pre_init_user(void) {
+    // If the audio is off during hardware init,
+    // it causes problem when the audio is turn on later.
+    // The workaround is to always set the audio on during hardware init,
+    // then toggle audio back to the setting.
+#ifdef AUDIO_ENABLE
+    if (!is_audio_on()) {
+        audio_disable = true;
+        audio_on();
+    }
+#endif
+}
+
+void matrix_init_user(void) {
+    // To Fix a audio related problem. See comments in keyboard_pre_init_user().
+#ifdef AUDIO_ENABLE
+    if (audio_disable) {
+        audio_off();
+    }
+#endif
+}
+
+void keyboard_post_init_user(void) {
+    rgblight_layers = rgb_layers;
+    rgblight_set_layer_state(0, true);
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t* record) {
+    switch (keycode) {
+        case CLEAR:
+            if (record->event.pressed) {
+                eeconfig_init();
+            }
+            return false;
+        default:
+            return true;
+    }
+}
+
+bool leader_found;
+LEADER_EXTERNS();
+
+void matrix_scan_user(void) {
+    LEADER_DICTIONARY() {
+        leading      = false;
+        leader_found = false;
+        SEQ_ONE_KEY(L_RESET) {
+            leader_found = true;
+            reset_keyboard();
+        }
+        else
+            SEQ_ONE_KEY(KC_DEL) {
+            leader_found = true;
+            layer_clear();
+        }
+        else
+            SEQ_ONE_KEY(LOWER) {
+            leader_found = true;
+            layer_on(_LOWER);
+        }
+        else
+            SEQ_ONE_KEY(RAISE) {
+            leader_found = true;
+            layer_on(_RAISE);
+        }
+        leader_end();
+    }
+}
+
+void leader_end(void) {
+    // Plays sound on if leader sequence found.
+    if (leader_found) {
+#ifdef AUDIO_ENABLE
+        PLAY_SONG(planck_sound);
+#endif
+    }
+}
+
+bool led_update_user(led_t led_state) {
+    // Turn on RBG for capslock.
+    rgblight_set_layer_state(4, led_state.caps_lock);
+    return true;
+}
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+    // Set RBG layer according to active keymap layer.
+    rgblight_set_layer_state(1, layer_state_cmp(state, 1));
+    rgblight_set_layer_state(2, layer_state_cmp(state, 2));
+    rgblight_set_layer_state(3, layer_state_cmp(state, 1) && layer_state_cmp(state, 2));
```

Comment:
The layers shouldn't be negative, as technically, it's asking for an unsigned integer.